LevelUp.Pos.ProposedOrders/ProposedOrderCalculator.cs

```diff
@@ -40,139 +40,166 @@ public override string ToString()
-            int customerPaymentAmount,
+            int outstandingAmount,
+            int taxAmount,
+            int exemptAmount,
+            int paymentAmount,
             int appliedDiscountAmount
         )
         {
-            int totalOutstandingAmountWithDiscount = totalOutstandingAmount + Math.Abs(appliedDiscountAmount);
+            int outstandingAmountWithDiscount = outstandingAmount + Math.Abs(appliedDiscountAmount);
 
-            int adjustedSpendAmount = CalculateAdjustedCustomerPaymentAmount(
-                customerPaymentAmount, 
-                totalOutstandingAmountWithDiscount);
+            return CalculateOrderValues(
+                outstandingAmountWithDiscount,
+                taxAmount,
+                exemptAmount,
+                paymentAmount);
+        }
 
-            int adjustedTaxAmount = CalculateAdjustedTaxAmount(
-                totalOutstandingAmountWithDiscount, 
-                totalTaxAmount,
-                adjustedSpendAmount);
+        internal static AdjustedCheckValues CalculateOrderValues(
+            int outstandingAmount,
+            int taxAmount,
+            int exemptionAmount,
+            int paymentAmount)
+        {
+            CheckData checkData = SanitizeData(outstandingAmount, taxAmount, exemptionAmount, paymentAmount);
+
+            var adjustedTaxAmount = CalculateAdjustedTaxAmount(checkData);
 
-            int adjustedExemptionAmount = CalculateAdjustedExemptionAmount(
-                totalOutstandingAmountWithDiscount,
-                totalTaxAmount,
-                totalExemptionAmount, 
-                adjustedSpendAmount);
+            var adjustedExemptionAmount = CalculateAdjustedExemptionAmount(checkData);
 
-            return new AdjustedCheckValues(adjustedSpendAmount, adjustedTaxAmount, adjustedExemptionAmount);
+            return new AdjustedCheckValues(checkData.PaymentAmount, adjustedTaxAmount, adjustedExemptionAmount);
         }
 
-        /// <summary>
-        /// A customer can not pay more than the largest amount due on a check.
-        /// </summary>
-        internal static int CalculateAdjustedCustomerPaymentAmount(int totalOutstandingAmount, int customerPaymentAmount)
+        internal static CheckData SanitizeData(int outstandingAmount, int taxAmount, int exemptionAmount, int paymentAmount)
         {
-            return Math.Max(0, Math.Min(customerPaymentAmount, totalOutstandingAmount));
+            var checkData = new CheckData
+            {
+                ExemptionAmount = exemptionAmount,
+                OutstandingAmount = outstandingAmount,
+                PaymentAmount = paymentAmount,
+                TaxAmount = taxAmount
+            };
+
+            checkData.PaymentAmount = PaymentAmountCannotBeGreaterThanOutstandingAmount(
+                checkData.OutstandingAmount, 
+                checkData.PaymentAmount);
+
+            checkData.TaxAmount = TaxAmountCannotBeLessThanOutstandingAmount(
```

Comment:
I think you mean `TaxAmountCannotBeGreaterThanOutstandingAmount`, because often times, tax is less than the outstanding amount.  I think if the tax amount is greater than the outstanding amount here, it is reducing the tax to be equal to the outstanding amount.